lib/base/dependencygraph.cpp

```diff
@@ -48,3 +49,27 @@ std::vector<Object::Ptr> DependencyGraph::GetParents(const Object::Ptr& child)
 
 	return objects;
 }
+
+std::set<Object::Ptr> DependencyGraph::GetAllParents(const std::vector<Object::Ptr>& children)
+{
+	std::set<Object::Ptr> objects, toDo (children.begin(), children.end());
+	std::shared_lock<std::shared_mutex> lock (m_Mutex);
+
+	do {
+		auto current (*toDo.begin());
```

Comment:
This assumes that `children` is never empty. Just using a regular `while (!toDo.empty())` loop would also just handle that case correctly.